examples/preprocessing/plot_power_transformer.py

```diff
@@ -0,0 +1,107 @@
+"""
+==========================================================
+Using PowerTransformer to apply the Box-Cox transformation
+==========================================================
+
+This example demonstrates the use of the Box-Cox transform through
+:class:`preprocessing.PowerTransformer` to map data from various distributions
+to a normal distribution.
+
+Box-Cox is useful as a transformation in modeling problems where
+homoscedasticity and normality are desired. Below are examples of Box-Cox
+applied to six different probability distributions: Lognormal, Chi-squared,
+Weibull, Gaussian, Uniform, and Bimodal.
+
+Note that the transformation successfully maps the data to a normal
+distribution when applied to certain datasets, but is ineffective with others.
+This highlights the importance ofvisualizing the data before and after
+transformation.
+"""
+
+# Author: Eric Chang <ericchang2017@u.northwestern.edu>
+# License: BSD 3 clause
+
+import numpy as np
+import matplotlib
+import matplotlib.pyplot as plt
+
+from sklearn.preprocessing import PowerTransformer, minmax_scale
+
+print(__doc__)
+
+
+N_SAMPLES = 3000
+
+pt = PowerTransformer(method='box-cox')
+rng = np.random.RandomState(304)
+size = (N_SAMPLES, 1)
+
+
+# lognormal distribution
+X_lognormal = rng.lognormal(size=size)
+
+# chi-squared distribution
+df = 3
+X_chisq = rng.chisquare(df=df, size=size)
+
+# weibull distribution
+a = 50
+X_weibull = rng.weibull(a=a, size=size)
+
+# gaussian distribution
+loc = 100
+X_gaussian = rng.normal(loc=loc, size=size)
+
+# uniform distirbution
+X_uniform = rng.uniform(low=0, high=1, size=size)
+
+# bimodal distribution
+loc_a, loc_b = 100, 105
+X_a, X_b = rng.normal(loc=loc_a, size=size), rng.normal(loc=loc_b, size=size)
+X_bimodal = np.concatenate([X_a, X_b], axis=0)
+
+
+# create plots
+distributions = [
+    ('Lognormal', X_lognormal),
+    ('Chi-squared', X_chisq),
+    ('Weibull', X_weibull),
+    ('Gaussian', X_gaussian),
+    ('Uniform', X_uniform),
+    ('Bimodal', X_bimodal)
+]
+
+colors = ['firebrick', 'darkorange', 'goldenrod',
+          'seagreen', 'royalblue', 'darkorchid']
+
+params = {
+    'font.size': 6,
+    'hist.bins': 150
+}
+matplotlib.rcParams.update(params)
```

Comment:
Is that a good idea? Depending on how careful ``sphinxgallery`` is with global state, I feel this could go wrong? Or for people copy and pasting the example?

Reply:
removed the global parameter setting